simpleclient_spring_boot_metric_writer/src/main/java/io/prometheus/client/spring/boot/SpringBootMetricWriter.java

```diff
@@ -0,0 +1,62 @@
+package io.prometheus.client.spring.boot;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Gauge;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.metrics.Metric;
+import org.springframework.boot.actuate.metrics.writer.Delta;
+import org.springframework.boot.actuate.metrics.writer.MetricWriter;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class SpringBootMetricWriter implements MetricWriter {
+  private static final String NAMESPACE = "spring_boot_metrics";
+
+  private final CollectorRegistry collectorRegistry;
+  private final ConcurrentMap<String, Gauge> gauges;
```

Comment:
This is a major anti-pattern, users should use Gauges directly rather than slowing things down and losing power by going via this indirection. I don't suppose there's a way to let users use Gauges/Counters directly?